### DISCUSSION TITLE
name 'serial' is not defined

### DISCUSSION THREAD

**Discussion by JimmyTang999** [pymodbus-dev/pymodbus#2200] [Q&A] [answered]
would someone here know in which python script serial is defined?

i am using code structure found [here](https://stackoverflow.com/questions/70686813/how-to-access-particular-registers-using-pymodbus-rtu) with a few adjustments to com port, baud and unit

**Comment by JimmyTang999** [comment]
had to pip install pyserial and then import serial

**Comment by janiversen** [comment, marked as answer]
Of course you have to install pyserial if you want to use serial ! please see our examples e,g, simple_sync.client,py
